ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.js

```diff
@@ -0,0 +1,349 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { toChecksumHexAddress } from '../../../../shared/modules/hexstring-utils';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { getMetaMaskAccounts } from '../../../selectors';
+import Button from '../../../components/ui/button';
+import CustodyLabels from '../../../components/institutional/custody-labels/custody-labels';
+import PulseLoader from '../../../components/ui/pulse-loader';
+import { INSTITUTIONAL_FEATURES_DONE_ROUTE } from '../../../helpers/constants/routes';
+import { SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP } from '../../../../shared/constants/swaps';
+import { CHAIN_IDS } from '../../../../shared/constants/network';
+import { shortenAddress } from '../../../helpers/utils/util';
+import Tooltip from '../../../components/ui/tooltip';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  mmiActionsFactory,
+  showInteractiveReplacementTokenBanner,
+} from '../../../store/institutional/institution-background';
+import Box from '../../../components/ui/box';
+import {
+  Text,
+  Label,
+  Icon,
+  ButtonLink,
+  IconName,
+  IconSize,
+} from '../../../components/component-library';
+import {
+  Color,
+  OVERFLOW_WRAP,
+  TextColor,
+  JustifyContent,
+  BLOCK_SIZES,
+  DISPLAY,
+  FLEX_DIRECTION,
+} from '../../../helpers/constants/design-system';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+
+const getButtonLinkHref = ({ address }) => {
+  const url = SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP[CHAIN_IDS.MAINNET];
+  return `${url}address/${address}`;
+};
+
+export default function InteractiveReplacementTokenPage({ history }) {
+  const dispatch = useDispatch();
+  const isMountedRef = useRef(false);
+  const mmiActions = mmiActionsFactory();
```

Comment:
I'm probably late to this party but why does this need to be a factory? Why can't we import the background actions individually as we do in other places?

Reply:
@zone-live will give you the historical context

Reply:
The factory pattern we have for our background actions came out because all the actions would have the same structure/logic, and since we can write that logic once in a function and use that function as a factory to all other functions/actions, it's just a way to not repeat ourselves so much.